### DISCUSSION TITLE
Where are attachment file stored ?

### DISCUSSION THREAD

**Discussion by nil0x42** [nocodb/nocodb#1417] [Q&A] [answered]
Hi, when i add a file to an attachment row, i can see that row content is a JSON containing URL to the file. But i would like to know if actual files content are stored as BLOBs in the database, or if they're stored somewhere else. I couldn't find documentation for this..

Thank you

**Comment by wingkwong** [comment, marked as answer]
If you haven't configured Storage in App Store. Then you should find them in ``NC_TOOL_DIR``.

**Reply by nil0x42** [reply]
Okay thank you, i'm now using a volume alias on docker in order to persist files.
